packages/components/src/snackbar/types.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import type { MutableRefObject, ReactNode, SyntheticEvent } from 'react';
+
+export type NoticeActionWithURL = {
+	label: string;
+	url: string;
+	onClick?: ( event: SyntheticEvent ) => void;
+};
+
+type NoticeActionWithOnClick = {
+	label: string;
+	url?: string;
+	onClick: ( event: SyntheticEvent ) => void;
+};
+
+export type NoticeAction = NoticeActionWithURL | NoticeActionWithOnClick;
+
+export type Notice = {
+	id: string;
+	spokenMessage: string;
+	actions: NoticeAction[];
+	icon?: ReactNode;
+	onDismiss?: () => void;
+	content: string;
+	isDismissible: boolean;
+	explicitDismiss: boolean;
+};
+
+export type SnackbarProps = {
+	/**
+	 * The displayed message of a notice. Also used as the spoken message for assistive technology, unless `spokenMessage` is provided as an alternative message.
+	 */
+	children: string;
+	/**
+	 * Used to provide a custom spoken message in place of the `children` default.
+	 *
+	 * @default children
+	 */
+	spokenMessage?: Notice[ 'spokenMessage' ];
+	/**
+	 * A politeness level for the notice's spoken message. Should be provided as one of the valid options for [an `aria-live` attribute value](https://www.w3.org/TR/wai-aria-1.1/#aria-live). Defaults to `"polite"`. Note that this value should be considered a suggestion; assistive technologies may override it based on internal heuristics.
+	 *
+	 * A value of `'assertive'` is to be used for important, and usually time-sensitive, information. It will interrupt anything else the screen reader is announcing in that moment.
+	 * A value of `'polite'` is to be used for advisory information. It should not interrupt what the screen reader is announcing in that moment (the "speech queue") or interrupt the current task.
+	 *
+	 * @default 'polite'
+	 */
```

Comment:
- no need to mention the default value in the description text, we already have a `default` tag
- let's not use markdown syntax for JSDoc links, instead we can use the `@see` notation
- let's keep each line within the 80 chars limit for improved legibility

```suggestion
	/**
	 * A politeness level for the notice's spoken message. Should be provided as
	 * one of the valid options for an `aria-live` attribute value. Note that this
	 * value should be considered a suggestion; assistive technologies may
	 * override it based on internal heuristics.
	 *
	 * A value of `'assertive'` is to be used for important, and usually
	 * time-sensitive, information. It will interrupt anything else the screen
	 * reader is announcing in that moment.
	 * A value of `'polite'` is to be used for advisory information. It should
	 * not interrupt what the screen reader is announcing in that moment
	 * (the "speech queue") or interrupt the current task.
	 *
	 * @see https://www.w3.org/TR/wai-aria-1.1/#aria-live
	 *
	 * @default 'polite'
	 */
```

Reply:
Good ideas, these lines were long.

Committed in https://github.com/WordPress/gutenberg/pull/45472/commits/70bfe595c8de58542f45de8ac4bb7b90fd31e75a